Removed unprincipled exercises and those implemented with mutability

raulraja opened the pull request:
The following PR gets rid of exercises that either use mutability or showed Scala techniques that may not be the best for newcomers getting introduced to the lang.
@dialelo @rafaparadela Can you please review? thanks!

dialelo commented on site/content/src/main/scala/stdlib/Implicits.scala:
```diff
@@ -102,12 +102,9 @@ object Implicits extends FlatSpec with Matchers with exercise.Section {
   def asDefaultImplicits(res0: Float, res1: Float) {
```
if we no longer use `res1` in the body of the exercise we can get rid of it.

dialelo commented on site/content/src/main/scala/stdlib/Implicits.scala:
```diff
@@ -116,13 +113,10 @@ object Implicits extends FlatSpec with Matchers with exercise.Section {
     def howMuchCanIMake_?(hours: Int)(implicit amount: BigDecimal, currencyName: String) =
       (amount * hours).toString() + " " + currencyName
 
-    implicit var hourlyRate = BigDecimal(34.00)
```
:fire: `res1`

dialelo commented on site/content/src/main/scala/stdlib/Lists.scala:
```diff
@@ -48,11 +48,12 @@ object Lists extends FlatSpec with Matchers with exercise.Section {
     */
   def headAndTailLists(res0: Int, res1: Int, res2: Int) {
     val a = List(1, 2, 3)
-    a.head should equal(res0)
```
we may want to encourage `Option` instead of `Some` for constructing optional values.

dialelo commented:
Awesome changes. I noticed some TODOs, are we addressing them as part of this pull request? I can create the appropiate issues if we're going to do it after merging this.

raulraja commented:
@dialelo Addressed all your comments. RE the TODOs I have already added the issues. Once is related to the compiler not being able to process multiline string literals and the other was related to the contravariance exercises which need to be rewritten without mutability.

dialelo commented:
Awesome, LGTM :shipit:
